src/commands/switch.js

```diff
@@ -0,0 +1,99 @@
+const { flags } = require('@oclif/command');
+const inquirer = require('inquirer');
+const AbstractAuthenticatedCommand = require('../abstract-authenticated-command');
+const BranchManager = require('../services/branch-manager');
+const ProjectManager = require('../services/project-manager');
+const withCurrentProject = require('../services/with-current-project');
+const envConfig = require('../config');
+
+class SwitchCommand extends AbstractAuthenticatedCommand {
+  async selectBranch(branches) {
+    try {
+      const { branch } = await inquirer.prompt([{
+        name: 'branch',
+        message: 'Select the branch you want to set current',
+        type: 'list',
+        choices: [
+          // NOTICE: Current branch should be last dispalyed branch.
+          ...branches.filter((currentBranch) => !currentBranch.isCurrent),
+          ...branches.filter((currentBranch) => currentBranch.isCurrent),
+        ].map((currentBranch) => currentBranch.name),
+      }]);
+
+      return branch;
+    } catch (error) {
+      const customError = BranchManager.handleBranchError(error);
+
+      return this.error(customError);
+    }
+  }
```

Comment:
I think this function can be put in an helper like `with-current-project`. it should be reused by the deploy command IMO.

Reply:
Thanks for this suggestion. At first glance, I thought I agree but since then I changed my mind. This method is currently specific to this case.

```js
message: 'Select the branch you want to set current',
```

```js
// NOTICE: Current branch should be last displayed branch.
```

We could consider avoiding a premature blind factorization: we will factorize (if required) when we need it, it would be better crafted. In other word, the developer that code the deploy command should not copy/paste if it's the exact same code: he should consider factorize it.

Reply:
Ok keep it like that we will see.